src/Atc.Cosmos.EventStore.Cqrs/Projections/ProjectionFactory.cs

```diff
@@ -16,4 +16,9 @@ public IProjection GetProjection<TProjection>()
     public IProjectionMetadata GetProjectionMetadata<TProjection>()
         where TProjection : IProjection
         => serviceProvider.GetRequiredService<ProjectionMetadata<TProjection>>();
+
+    public ProjectionFactory CreateScope()
+    {
+        return new ProjectionFactory(serviceProvider.CreateScope().ServiceProvider);
+    }
```

Comment:
Remember to dispose the scope otherwise scoped IDisposable services resolved through the scope will not be disposed

Reply:
To keep explicit control of how a scope is created and disposed, I have decided to inject IServiceProvider and make use of it instead of returning the scope from the factory. 
Factory was singleton, and now makes very little sense, thus I have remove it and call GetService directly on the scoped IServiceProvider.